web/api/v1/api.go

```diff
@@ -286,6 +295,16 @@ func NewAPI(
 	return a
 }
 
+// InstallCodec adds codec to this API's available codecs.
+// If codec handles a content type handled by a codec already installed in this API, codec replaces the previous codec.
+func (api *API) InstallCodec(codec Codec) {
+	if api.codecs == nil {
+		api.codecs = map[string]Codec{}
+	}
```

Comment:
[nit] Can be a real case having `api.codes = nil`?

Reply:
In production code: no. This made some of the tests simpler. 

There are some tests that don't use `NewAPI()` to create the `API` instance but instead create a `API` instance directly, so `codecs` is `nil`. Initialising `codecs` in the tests felt a bit weird, I thought the tests shouldn't have knowledge of this internal detail.

Changing these tests to use `NewAPI()` is certainly possible but seemed like overkill for what they were testing.